scripts/BuildAndTest.ps1

```diff
@@ -200,7 +200,7 @@ if (-not $NoBuildSample) {
 }
 
 if (-not $NoTest) {
-    & $PSScriptRoot\Run-Tests.ps1
+    & $PSScriptRoot\Run-Tests.ps1 $Configuration
```

Comment:
I suspect you either need to specify the -Configuration switch or add the Position attribute to the param in Run-Tests.ps1.

Reply:
This definitely invokes the debug tests. It's all academic, however, as the debug builds don't pass tests. Likely need to scrub for asserts that are firing.

Reply:
Yes and no.

1. By default `Run-Tests.ps1` runs the `Release` tests, not the `Debug` tests. See the declaration of the `$Configuration` parameter in `Run-Tests.ps1` on line 21.
2. It is definitely a bug that this invocation of `Run-Tests.ps1` did not pass the `$Configuration` value from the calling script. Thanks for fixing.
3. Even though `$Configuration` is a positional parameter, and the first one (so that the change you made works properly), I always prefer to spell out the parameter name, thus:
```
& $PSScriptRoot\Run-Tests.ps1-Configuration $Configuration
```
This is good documentation (although admittedly of limited value in this particular case), and it makes the caller resilient against changes to the parameter declaration order in the callee.

---
In reply to: [212344747](https://github.com/Microsoft/sarif-sdk/pull/1044#discussion_r212344747) [](ancestors = 212344747)